### ISSUE TITLE
Synchronize deck.gl and luma.gl versions with `vitessce`

### ISSUE THREAD

**Issue by NickAkhmetov** [hubmapconsortium/ccf-ui#1146]
## Background

The HuBMAP Data Portal uses Vitessce to visualize individual datasets' bioimaging data. On organ pages, Vitessce is used in conjunction with the CCF Organ Info component to display a reference dataset from Azimuth, e.g. https://portal.hubmapconsortium.org/organ/kidney

In order to have both components mounted on the page at once, we currently display the CCF UI organ info component in an iframe. This has several drawbacks in terms of both performance and data flow, so we are trying to move away from this approach.

## Problem

Attempting to mount both Vitessce and CCF UI within the same JS context (i.e. both components in the same HTML document without using an iframe) reveals minor mismatches of `deck.gl` and `luma.gl` versions; Vitessce uses deck.gl 8.8 (the package is pinned to ~8.8.6, which resolves to 8.8.20) and luma.gl 8.5.16, while the Organ Info web components ship with deck.gl 8.5.8 / luma.gl 8.5.12. This mismatch leads to thrown runtime errors:

<details><summary>deck.gl multiple versions detected</summary>
<p>

![image](https://github.com/hubmapconsortium/ccf-ui/assets/19957804/572403b9-5c98-4dcb-996e-0a824b5adfba)

</p>
</details> 

<details><summary>luma.gl - multiple VERSIONs detected</summary>
<p>

![image](https://github.com/hubmapconsortium/ccf-ui/assets/19957804/080729f2-0971-4089-ba23-d677ba5a68ec)

</p>
</details> 

## deck.gl 8.9

When this issue arose, I spoke with @bherr2 about this on Slack; given that there was already a planned upgrade to deck.gl 8.9.22 and luma.gl 8.5.20, we decided that bumping up to the latest stable version of deck.gl (8.9.30) and luma.gl (8.5.21) would be the best way to synchronize dependency versions, so I started working on upgrading those dependencies on the Vitessce side: https://github.com/vitessce/vitessce/pull/1718

Mark and I soon discovered there are issues with Deck.gl and Luma's packaging in v8; in essence, due to the ESModule distribution of Deck and Luma not being properly indicated by either a `.mjs` extension or an `exports` field in the package, node always resolves to use the CJS distribution. Up until Deck.gl 8.9, this was not an issue, as all of its subdependencies shipped with a CJS entry point; however, Deck.gl 8.9 included an upgrade to the new version of `mapbox`, which now only provides an ESM entrypoint. This led to Vitessce's test suite breaking due to the ESM resolution failing; though we were able to work around this for the test suite via `pnpm patch`, we realized that releasing a build of Vitessce with this deck.gl version would lead to other complications for other users, e.g. anyone using server-side rendering frameworks.

Mark filed related issues in the deck.gl repository:
- https://github.com/visgl/deck.gl/issues/8186
- https://github.com/visgl/deck.gl/issues/8187

To summarize the [most complete response](https://github.com/visgl/deck.gl/issues/8187#issuecomment-1755643069):
- v9 will resolve the ES5/ESM packaging issue
- v8.9 is a transition version that was released to patch security issues in dependencies
- **For now we should probably stay on 8.8**

## Conclusion

In short, setting the versions of `deck.gl` and `luma.gl` to match Vitessce's current versions appears to be the most straightforward route to ensure compatibility between them in the short term.

**Comment by bherr2** [comment]
@NickAkhmetov what exact versions would you like to set for deck.gl and luma.gl? I can add this to our staging site today and you can test it out tomorrow if you like.

**Comment by NickAkhmetov** [comment]
The current versions in the lockfile are:
- deck.gl 8.8.20 (~8.8.6)
- luma.gl 8.5.16 
- loaders.gl 3.2.10 (in case it's also relevant!)

**Comment by bherr2** [comment]
@NickAkhmetov I'm getting
```
Uncaught Error: luma.gl - multiple VERSIONs detected: 8.5.16 vs 8.5.21
  Error: luma.gl - multiple VERSIONs detected: 8.5.16 vs 8.5.21
```

When I set the versions to specifically 8.5.16 luma.gl ... Any thoughts?

**Comment by NickAkhmetov** [comment]
I'm getting the same 8.5.16 vs 8.5.21 error you reported. I think we may be able to bump Vitessce to that luma version without bumping the deck.gl version; I'll test it out and let you know.

**Comment by NickAkhmetov** [comment]
@bherr2 I bumped `vitessce` up to `luma.gl@8.5.21` and `deck.gl@8.8.27`; the staging build of the organs component and the local build of that vitessce component successfully mount on the same page! 🙌🏻  https://github.com/vitessce/vitessce/pull/1724

![image](https://github.com/hubmapconsortium/ccf-ui/assets/19957804/1f7f7723-7a4b-4d7f-9a62-36d876a2acd9)

**Comment by bherr2** [comment]
Awesome!

**Comment by bherr2** [comment]
Can we close this? Once we publish, if it needs changed again, just open a new issue and we can do that pretty quick.

**Comment by NickAkhmetov** [comment]
This should be all set 👍🏻 Thank you! I'll close this as resolved.
